go/database/mpt/io/archive.go

```diff
@@ -227,7 +227,11 @@ func importArchive(liveDbDir, archiveDbDir string, in io.Reader) (err error) {
 	if _, err := io.ReadFull(in, buffer); err != nil {
 		return err
 	} else if !bytes.Equal(buffer, archiveMagicNumber) {
-		return fmt.Errorf("invalid format, wrong magic number")
+		// Specify error if incorrect genesis is passed
+		if bytes.Contains(buffer, stateMagicNumber) {
+			return fmt.Errorf("incorrect genesis+command combination\n your genesis is meant to be used with import-live")
+		}
```

Comment:
I would like to avoid the term `genesis` here, to reduce the confusion with what is considered a `genesis` file in the full Sonic client. What we are dealing with here is only a small part of it, and so far I think we are not referring to it in the code as `genesis`.

Suggestion:
```suggestion
		// Provide an explicit warning to the user if instead of an archive dump a live-db dump was provided
		if bytes.Contains(buffer, stateMagicNumber) {
			return fmt.Errorf("incorrect input data format; use the `import` or `import-live-db` sub-command with this type of data")
		}
```